doc/en/user/source/community/metadata/uiconfiguration.rst

```diff
@@ -0,0 +1,359 @@
+
+ .. figure:: images/fielduuid.png
+
+
+
+.. code:: YAML
+
+  attributes:
+    - key: uuid-field
+      fieldType: UUID
+
+NUMBER
+^^^^^^
+Only numbers are accepted as valid input.
+
+ .. figure:: images/fieldnumber.png
+
+
+
+.. code:: YAML
+
+  attributes:
+    - key: number-field
+      fieldType: NUMBER
+
+BOOLEAN
+^^^^^^^
+Input field with checkbox.
+
+ .. figure:: images/fieldboolean.png
+
+
+
+.. code:: YAML
+
+  attributes:
+    - key: boolean-field
+      fieldType: BOOLEAN
+
+DATE
+^^^^
+
+Date selection without time information.
+
+ .. figure:: images/fielddate.png
+
+
+
+.. code:: YAML
+
+  attributes:
+    - key: date-field
+      fieldType: DATE
+
+
+DATETIME
+^^^^^^^^
+
+Selection date with time information.
+
+ .. figure:: images/fielddatetime.png
+
+
+
+.. code:: YAML
+
+  attributes:
+    - key: datetime-field
+      fieldType: DATETIME
+
+DROPDOWN
+^^^^^^^^
+A component for selecting a value from a dropdown. 
+The values can be configured with the ``values`` attribute in the yaml or they can be configured in an other .csv file which is used for dropdowns with a lot of choices.
+
+
+ .. figure:: images/fielddropdown.png
+
+
+Configuration in the yaml file.
+
+.. code:: YAML
+
+  attributes:
+    - key: dropdown-field
+      fieldType: DROPDOWN
+      values:
+            - first
+            - second
+            - third
+
+To configure the values in a separate file add a yaml key ``csvImports`` on the same level as ``attributes`` and add the list of CSV files under this key. 
+The first line in each CSV file should contain the key of the dropdown component for which you want to add the choices.
+
+``metadata-ui.yaml``
+
+.. code:: YAML
+
+  attributes:
+    - key: dropdown-field
+      fieldType: DROPDOWN
+   csvImports:
+    - dropdowncontent.csv   
+        
+``dropdowncontent.csv``
+
+.. code::
+
+    dropdown-field
+    first
+    second
+    third
+
+SUGGESTBOX
+^^^^^^^^^^
+A component for selecting a value from a suggestbox. Suggestions will be given for the values where the input matches the beginning of the possible values.
+The values can be put in a separate CSV file in the same way as for the DROPDOWN component. 
+
+.. figure:: images/fieldsuggest.png
+
+.. code:: YAML
+
+  attributes:
+    - key: suggestbox-field
+      fieldType: SUGGESTBOX
+      values:
+            - first
+            - second
+            - third
+
+DERIVED
+^^^^^^^
+A derived field is a hidden field whose value depends on an other component. The yaml key ``derivedFrom`` should contain the key of the component it depends on.
+When a value is selected in the parent component a matching value for the derived component is searched in csv file.
+
+The CSV file should start wiht the key of the parent component as first column and the second column should contain the key of the derived component. The following lines contain the mapping between the parent value and the derived value.
+
```

Comment:
In principe kan zo een CSV ook meer dan twee kolommen bevatten, wat je nodig hebt als je meer dan een veld zou afleiden van ehtzelfde veld